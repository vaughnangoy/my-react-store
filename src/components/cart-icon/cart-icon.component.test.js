import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import CartIcon from './cart-icon.component';

describe('Cart Icon', () => {
    afterEach(cleanup);

    it('renders', () => {
        const { asFragment } = render(<CartIcon />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should initialise the counter to 0', () => {
        const { getByTestId } = render(<CartIcon />);
        expect(getByTestId('counter')).toHaveTextContent('0');
    });

    it('should update the counter to 3', () => {
        const { getByTestId, rerender } = render(<CartIcon />);
        expect(getByTestId('counter')).toHaveTextContent('0');
        rerender(<CartIcon counter={3} />);
        expect(getByTestId('counter')).toHaveTextContent('3');
    });
});
