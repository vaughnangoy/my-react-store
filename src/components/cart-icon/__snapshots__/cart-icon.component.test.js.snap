// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Icon renders 1`] = `
<DocumentFragment>
  <div
    class="cart-icon"
  >
    <svg
      class="shopping-icon"
    >
      shopping-bag.svg
    </svg>
    <span
      class="item-count"
      data-testid="counter"
    >
      0
    </span>
  </div>
</DocumentFragment>
`;
